### ISSUE TITLE
[JENKINS-12063] Automatically update graphs when job finishes

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#11404]
<p>You are on a job page, the job is in progress, regardless of whether there is compiler warnings graph (2+ jobs run before) or no graph (only one job run before), when the job finishes, the build status updates but not the graph.</p>

<p>Trend graphs are represented by static HTML in the moment. They will be updated only if the page is refreshed (see refresh interval). </p>

---
<details><summary><i>Originally reported by <a title="Greg Moncreaff" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=moncreaff">moncreaff</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-12063" target="_blank">Automatically update graphs when job finishes</a></i></summary>
<i><ul>
<li><b>reported by</b>: moncreaff
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2021-05-06T21:36:56+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-07
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=moncreaff">moncreaff</a>:</i>
<p>Is the build history not gated by refresh delay?  Should it trigger a refresh regardless of the delay when job finishes, or are these to things completely independent of each other?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=moncreaff">moncreaff</a>:</i>
<p>you may need to set your refresh rate to some large (multi 100) number of seconds to see what I mean</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli">drulli</a>:</i>
<p>Hmm, I'm not sure if I understand correctly (the comments and the title seem to be different): does the warnings graph update after the auto-refresh interval or not?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=moncreaff">moncreaff</a>:</i>
<p>Sorry for the confusion.  Yes it does update after the refresh interval.  I guess what I was trying to say is that regardless of the interval, if the build is being observed and the build progress changes to completed, should that trigger an immediate refresh?  </p>

<p>CONOPS is to set long refresh interval so pages aren't polled regardless of change, but when something interesting happens, page is auto updated then.</p>
